src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -153,7 +162,7 @@ uct_cuda_ipc_mem_add_reg(void *addr, uct_cuda_ipc_memh_t *memh,
                         CU_MEM_HANDLE_TYPE_FABRIC, 0));
         if (status != UCS_OK) {
             cuMemRelease(handle);
-            goto non_ipc;
+            goto err;
```

Comment:
Wasn't that an original intent to fallback if `cuMemExportToShareableHandle` fails?

Reply:
We shouldn't be hitting this case if the conditions are already met here https://github.com/openucx/ucx/pull/10189/files#diff-91e9ff28493264cecd03ffb88c50c0af9ff1be814254383dcec6ec40814d0f80R152. And if allocation doesn't have a fabric handle associated with it, then we will end up going to `non_ipc` path.

Reply:
discussed offline